widgetastic_manageiq/__init__.py

```diff
@@ -4437,3 +4437,150 @@ def fill(self, value):
+
+    def is_datastores_expanded(self, map_name):
+        el = self._get_map_element(map_name)
+        datastores_button_el = self.browser.element(self.ITEM_DATASTORES_LOCATOR, parent=el)
+        return len(self.browser.elements(self.ITEM_EXPAND_LOCATOR, parent=datastores_button_el)) > 0
+
+    def is_associated_plans_expanded(self, map_name):
+        el = self._get_map_element(map_name)
+        associated_plans_button_el = self.browser.element(
+            self.ITEM_ASSOCIATED_PLANS_BUTTON_LOCATOR, parent=el)
+        return len(self.browser.elements(self.ITEM_EXPAND_LOCATOR,
+            parent=associated_plans_button_el)) > 0
+
+    def _get_map_element(self, map_name):
+        for item in self.browser.elements(self.ITEM_LOCATOR):
+            el = self.browser.element(".//*[@class='list-group-item-heading']", parent=item)
+            if self.browser.text(el) == map_name:
+                return item
+        raise ItemNotFound("Infra Mapping: {} not found".format(map_name))
+
+    def expand_map_clusters(self, map_name):
+        """Click on clusters to expand to show details."""
+        el = self._get_map_element(map_name)
+        self.browser.click(self.browser.element(self.ITEM_CLUSTERS_LOCATOR, parent=el))
+
+    def expand_map_networks(self, map_name):
+        """Click on networks to expand to show details."""
+        el = self._get_map_element(map_name)
+        self.browser.click(self.browser.element(self.ITEM_NETWORKS_LOCATOR, parent=el))
+
+    def expand_map_datastores(self, map_name):
+        """Click on datastores to expand to show details."""
+        el = self._get_map_element(map_name)
+        self.browser.click(self.browser.element(self.ITEM_DATASTORES_LOCATOR, parent=el))
+
+    def expand_map_associated_plans(self, map_name):
+        """Click on associated plans to expand to show details."""
+        try:
+            el = self._get_map_element(map_name)
+            self.browser.click(self.browser.element(
+                self.ITEM_ASSOCIATED_PLANS_BUTTON_LOCATOR, parent=el))
+        except NoSuchElementException:
+            raise ItemNotFound("There are no associated plans with map {}".format(map_name))
+
+    def read(self):
+        return self.all_items
+
+    def get_map_status(self, map_name):
+        el = self._get_map_element(map_name)
+        return self.browser.text(self.browser.element(self.ITEM_STATUS_LOCATOR,
+             parent=el))
+
+    def _get_sources(self, map_name):
+        el = self._get_map_element(map_name)
+        for item in self.browser.elements(self.SOURCE_ITEMS_LIST_LOCATOR, parent=el):
+            return self.browser.text(item)
+
+    def _get_targets(self, map_name):
+        el = self._get_map_element(map_name)
+        for item in self.browser.elements(self.TARGET_ITEMS_LIST_LOCATOR, parent=el):
+            return self.browser.text(item)
+
+    def get_map_source_clusters(self, map_name):
+        if not self.is_clusters_expanded(map_name):
```

Comment:
I think you should bake all of these checks into the `expand_map_<thing>` function calls, so anything trying to expand the element doesn't have to check first.

Also, these need docblocks.